OneCommunityGlobal/HGNRest
# Abdel deadline follow up tracking solution

# Description
1.	#### $\textcolor{red}{\textsf{(PRIORITY HIGH) Jae:}}$  Create way for me to track people I’ve followed up with about deadlines 

-  On the Tasks tab, I’d like a little box I can click on and off. I’m thinking the best location is to the right of the time numbers below. This is because it is related to those numbers and I want it super easy to scroll the whole tasks list and see who needs a followup or not.
- It should be visible and usable by every class but Volunteers. Any class checking the box will change its appearance for all others that can see it. This is so everyone knows if a person has been followed up with yet.
- It should have mouseover text that says, “This box is used to track follow ups. Clicking it means you’ve checked in with a person that they are on track to meet their deadline.”
- I’d like the box to be Red outline with light pink filler when a person is at 50% of their task deadline, again at 75% of their task deadline, and again when they are at 90% of their task deadline. This is to make it clear that it needs to be checked/the person needs to be followed up with. Once checked, the box should change back to normal but with a check in it. 
- I’d like the box to reset/auto-clear the check within the box when a person reaches 75% and 90%, so the box is ready/needs to be checked again. This is so I can track when I’ve followed up with people and also know when/if my managers are following up with people.
- I’d like an “i” icon for information that explains all of the information above.
- I’d also like mouseover text that will be different for the box at each level 
      **A -**. 0-50%: Check this box once you’ve checked in for the first time with this team member to make sure they are clear on their task.
      **B -**. 0-50%: Check this box once you’ve checked in for the first time with this team member to make sure they are clear on their task.
      **C -**. Your team member’s task should be at least 75% complete! Check this box once you’ve confirmed they are on track to meet their deadline. Request additional time be added to their task if it is needed.
      **D -**. Your team member’s task should be almost complete! Check this box once you’ve confirmed they are on track to meet their deadline. Request additional time be added to their task if it is needed.
![image](https://github.com/OneCommunityGlobal/HGNRest/assets/49083865/b40da0a9-c774-4cd5-a884-c9c415506d81)

## Related PRS :

- To test this backend PR you need to checkout the **935** frontend PR.

## Main changes explained:

- Update taskController file Add setFollowUp.
- Update taskRoutes file Add new route for setFollowUp.
- Update task model file Add followedUp


## How to test:

- check into current branch
- do `npm run build`
- do `npm start`
- checkout the **935** frontend PR.
